scripts/process/sessions_all.R

```diff
@@ -1,82 +1,146 @@
+
+  break_data <- filter(summary_data, grepl(break.by, type)) %>%
+    mutate(bin = cut(sessions,
                      breaks = c(-Inf, viz[['breaks']], Inf),
-                     labels = c("Low", "Moderate", "High", "Very High") )) %>% 
+                     labels = c("Low", "Moderate", "High", "Very High") )) %>%
     arrange(desc(sessions))
-  
 
-  summary_data_full <- left_join(summary_data, 
+
+  summary_data_full <- left_join(summary_data,
                                  select(break_data, bin, longName), by="longName")%>%
     mutate(longName = factor(longName, levels = rev(break_data$longName)),
            bin = factor(bin, levels = c("Very High", "High", "Moderate", "Low")),
+           type = factor(type, levels = level_text),
            scaler = 1)
-  
+
   scale_to <- max(summary_data_full$sessions, na.rm = TRUE)
-  
+
   scaler_df <- summary_data_full %>%
     group_by(bin) %>%
     summarize(max_bin = max(sessions, na.rm = TRUE)) %>%
     data.frame()
-  
+
   summary_data_full <- summary_data_full %>%
     left_join(scaler_df, by = "bin") %>%
     mutate(scaler = scale_to / max_bin,
-           scaled_value = sessions * scaler, 
+           scaled_value = sessions * scaler,
            scaled_newUser = newUsers * scaler) %>%
     select(-max_bin, -scaler)
-  
+
+  #Find sites with data in one but not the other bin/type:
+
+  counts <- data.frame(t(table(summary_data_full[,c("bin","type")]))) %>%
+    spread(type, Freq)
+
+  find_zeros <- counts %>%
+    select(-bin)
+
+  find_zeros$any_gone <- apply(find_zeros, 1, function(x) length(unique(x)) == 1)
+
+  find_zeros <- find_zeros %>%
+    bind_cols(bin = counts$bin) %>%
+    filter(!any_gone) %>%
+    select(-any_gone)
+
+  if(nrow(find_zeros) > 0){
+
+    find_maxes <- gather(find_zeros, type, value, -bin) %>%
+      group_by(bin) %>%
+      top_n(n=1)
+
+    # bins and key's we need to add 0's:
+    find_zeros_long <- gather(find_zeros, type, value, -bin) %>%
+      anti_join(find_maxes) %>%
+      left_join(distinct(select(find_maxes, bin, max = value))) %>%
+      mutate(num_sites = max - value)
+
+    find_zeros_long$type <- factor(find_zeros_long$type, levels = level_text)
+
+    #Names in maxes not in zero's:
+    for(j in 1:nrow(find_zeros_long)){
+
+      sites_to_add <- select(summary_data_full, longName, shortName, type, bin) %>%
+        filter(bin == find_zeros_long$bin[j])%>%
+        filter(type != find_zeros_long$type[j]) %>%
+        select(-type,-bin) %>%
+        distinct()
+
+      sites_already_there <- select(summary_data_full, longName, shortName, type, bin) %>%
+        filter(bin == find_zeros_long$bin[j])%>%
+        filter(type == find_zeros_long$type[j]) %>%
+        select(-type,-bin) %>%
+        distinct()
+
+      new_sites <- anti_join(sites_to_add, sites_already_there)
+
+      add_missing <- cbind(new_sites,
+                           find_zeros_long[j,c("bin","type")])
+
+      add_missing <- cbind(add_missing,
```

Comment:
bind_cols would take a list of things to bind, could reduce # of calls from 2 to 1 in L127-136